packages/admin/admin/src/stack/breadcrumbs/StackBreadcrumbs.tsx

```diff
@@ -1,46 +1,63 @@
-                    <MuiBreadcrumbs
-                        classes={{ root: classes.root, ol: classes.ol, li: classes.li, separator: classes.separator }}
-                        separator={separator}
-                        {...otherProps}
-                    >
-                        {stackApi?.breadCrumbs.map(({ id, url, title }, index) => {
-                            const isLast = index + 1 >= stackApi?.breadCrumbs.length;
-                            const linkClassName = classes.link + (isLast ? ` ${classes.last}` : "");
+}: StackBreadcrumbsProps & WithStyles<typeof styles>): React.ReactElement | null => {
+    const stackApi = useStackApi();
+    const { palette } = useTheme();
+    const breadcrumbsRef = React.useRef<HTMLElement>(null);
+    const [breadcrumbsResizeListener, { width: breadcrumbsContainerWidth }] = useResizeAware();
+
+    const numberOfItemsToBeHidden = useNumberOfItemsToBeHidden(
+        breadcrumbsRef.current,
+        breadcrumbsContainerWidth,
+        stackApi?.breadCrumbs ? stackApi.breadCrumbs.length : 0,
+        itemsBeforeCollapse,
+    );
 
-                            return (
-                                <Link key={id} to={url} component={BreadcrumbLink} className={linkClassName}>
-                                    <Typography variant="body2">{title}</Typography>
-                                </Link>
-                            );
-                        })}
-                    </MuiBreadcrumbs>
-                );
-            }}
-        </StackApiContext.Consumer>
+    if (!stackApi?.breadCrumbs?.length) return null;
+
+    const breadcrumbItemsBeforeCollapse = stackApi.breadCrumbs.slice(0, itemsBeforeCollapse);
+    const breadcrumbItemsAfterCollapse = stackApi.breadCrumbs.slice(itemsBeforeCollapse, stackApi.breadCrumbs.length);
+    const breadcrumbItemsInsideOverflowMenu = breadcrumbItemsAfterCollapse.splice(0, numberOfItemsToBeHidden ?? 0);
+
+    return (
+        <div className={classes.root}>
+            {breadcrumbsResizeListener}
+            <MuiBreadcrumbs
+                ref={breadcrumbsRef}
+                maxItems={stackApi.breadCrumbs.length + 1} // Prevent the default collapse behavior to use our own overflow menu.
+                classes={{ root: classes.breadcrumbs, ol: classes.ol, li: classes.li, separator: classes.separator }}
+                separator={separator === undefined ? <ChevronRight fontSize="inherit" htmlColor={palette.grey[900]} /> : separator}
+                {...otherProps}
+            >
+                {getBreadcrumbLinkNodes(
+                    breadcrumbItemsBeforeCollapse,
+                    breadcrumbItemsBeforeCollapse.length === stackApi.breadCrumbs.length,
+                    classes,
+                    stackApi.breadCrumbs.length > 1 ? stackApi.breadCrumbs[stackApi.breadCrumbs.length - 2].url : undefined,
+                )}
+                {(breadcrumbItemsInsideOverflowMenu.length || numberOfItemsToBeHidden === null) && (
```

Comment:
```suggestion
                {(breadcrumbItemsInsideOverflowMenu.length > 0 || numberOfItemsToBeHidden === null) && (
```

Using a falsy check here might result in rendering a 0

Reply:
Not really, the expression `(breadcrumbItemsInsideOverflowMenu.length || numberOfItemsToBeHidden === null)` can only result in a boolean.